source/simple_sub_pub_demo.c

```diff
@@ -193,6 +197,28 @@ static void prvSimpleSubscribePublishTask( void * pvParameters );
  */
 extern MQTTAgentContext_t xGlobalMqttAgentContext;
 
+/**
+ * @brief The global array of subscription elements.
+ *
+ * @note No thread safety is required to this array, since the updates the array
+ * elements are done only from one task at a time.
+ */
+extern SubscriptionElement_t xGlobalSubscriptionList[ SUBSCRIPTION_MANAGER_MAX_SUBSCRIPTIONS ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The buffer to hold the topic filter. The topic is generated at runtime
+ * by adding the task names.
+ *
+ * @note The topic strings must persist until unsubscribed.
+ */
+#if democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE > 0
+    static char topicBuf[ democonfigNUM_SIMPLE_SUB_PUB_TASKS_TO_CREATE ][ mqttexampleSTRING_BUFFER_LENGTH ];
+#else
+    static char topicBuf[ 1U ][ mqttexampleSTRING_BUFFER_LENGTH ];
+#endif
```

Comment:
I don't get this. This buffer was previously part of each task's stack, and making it static only moves it from there to the application's static memory, so it doesn't save anything. Is there any reason to make this static?

Reply:
Yes. It was in application's stack previously, but that model will not work with this subscription manager implementation. The topic Filters needs to be in memory for the subscription manager implementation until these topics are unsubscribed. So this change is required.